### PR TITLE
Fix/improve TCP option parsing

### LINKED ISSUE #618: Read TCP option values
It would be nice if the tcp_options event (https://docs.zeek.org/en/stable/scripts/base/bif/plugins/Zeek_TCP.events.bif.zeek.html#id-tcp_option )
would allow the code to read the value of the given TCP option.

### PR DESCRIPTION
**Pull request by jsiwek**
Merge the same branch name in `zeek-docs` along with this.

Fixes #618 

And unless there's objections, I'll pick 222e3ad3ea82ee212717d103e1bfc718f11256eb into `release/3.0` since the original `tcp_option` event is otherwise broken.

### CODE REVIEW

File: src/analyzer/protocol/tcp/TCP.cc [hunk 1]
```diff
@@ -1307,47 +1305,132 @@ int TCP_Analyzer::ParseTCPOptions(const struct tcphdr* tcp,
...
+
+			switch ( kind ) {
+			case 2:
+				// MSS
+				if ( length == 4 )
+					{
+					auto mss = ntohs(*reinterpret_cast<const uint16_t*>(o + 2));
+					option_record->Assign(3, val_mgr->GetCount(mss));
+					handled_option = true;
+					}
```
Comment (0xxon) [inline src/analyzer/protocol/tcp/TCP.cc]: Would it perhaps make sense to raise weird events in cases where the length does not match what we expect? I know we do still report them - but we don't make it that easy to show that something potentially weird is up.

**Comment by 0xxon** [comment]
Just added a tiny question before merging this.

File: src/analyzer/protocol/tcp/TCP.cc (hunk 1 above)
Comment (jsiwek) [reply src/analyzer/protocol/tcp/TCP.cc]: Yeah, seems that could be helpful; pushed a new commit to add those weirds.

**Comment by 0xxon** [comment]
Merged. TBH, I am not really sure that I feel this is important to be merged into 3.0.1 - especially if the tcp_option event has been broken for a time (so it is not a regression).

**Comment by jsiwek** [comment]
Generally, I like the idea of merging into the release branch any fixes to bugs or regressions unless there's obvious workarounds a user can do or unless an unreasonable amount of changes are required.  In this case, it's a simple bug that makes `tcp_option` an unusable event with no workaround, but since it's not exactly an event that carries the most usable information in the first place, I'm fine scratching it from the `release/3.0` list unless someone else wants to vote it in.

**Comment by 0xxon** [comment]
I actually don't have a very strong opinion about this one - I can see why merging it would be neat. It just also doesn't feel super urgent. So - potentially just ignore me :)

**Pull request merged**